clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -4160,6 +4178,8 @@ llvm::DISubprogram *CGDebugInfo::getFunctionFwdDeclOrStub(GlobalDecl GD,
   if (Stub) {
     Flags |= getCallSiteRelatedAttrs();
     SPFlags |= llvm::DISubprogram::SPFlagDefinition;
+    if (usesDebugTransparent(FD, CGM))
```

Comment:
Do we really need this?

AFAICT, the stub is only generated when we do `EmitInlinedInheritingCXXConstructorCall`. Those inheriting constructors don't support the `debug_transparent` attribute anyway. I think it would be sufficient to just add a test that ensures we do the right thing for inlined functions.

Reply:
Turns out adding this flag only in `emitFunction` is enough to cover all cases.